editor/components/page-attributes/check.js

```diff
@@ -26,30 +21,21 @@ export function PageAttributesCheck( { availableTemplates, postType, children }
 	return children;
 }
 
-const applyConnect = connect(
-	( state ) => {
-		return {
-			postTypeSlug: getCurrentPostType( state ),
-		};
-	}
-);
+const applyWithSelect = withSelect( ( select ) => {
+	const { getEditedPostAttribute } = select( 'core/editor' );
+	const { getPostType } = select( 'core' );
+	return {
+		postType: getPostType( getEditedPostAttribute( 'type' ) ),
```

Comment:
General observation. It looks like `getCurrentPostType` should be deprecated and replaced with `getEditedPostAttribute` in all other places with the follow-up PRs.